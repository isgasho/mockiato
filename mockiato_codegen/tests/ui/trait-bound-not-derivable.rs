#![feature(plugin)]
#![plugin(mockiato_codegen)]

trait Bar {}

#[mockable]
trait Foo: Bar + std::fmt::Debug {}

#[mockable]
trait Baz: std::fmt::Display {}
