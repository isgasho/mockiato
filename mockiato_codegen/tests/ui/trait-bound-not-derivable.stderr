error: The referenced trait is not a derivable trait
 --> $DIR/trait-bound-not-derivable.rs:7:12
  |
7 | trait Foo: Bar + std::fmt::Debug {}
  |            ^^^
  |
  = help: Only traits that are automatically derivable are supported as supertrait

error: The referenced trait is not a derivable trait
  --> $DIR/trait-bound-not-derivable.rs:10:12
   |
10 | trait Baz: std::fmt::Display {}
   |            ^^^^^^^^^^^^^^^^^
   |
   = help: Only traits that are automatically derivable are supported as supertrait

error: aborting due to 2 previous errors

